Resources/config/services.xml

```diff
@@ -1,61 +1,63 @@
-            <argument type="service" id="Symfony\Component\Cache\Adapter\AdapterInterface"/>
+        <service id="Shivas\VersioningBundle\Service\VersionManager" public="true">
+            <argument type="service" id="shivas_versioning.cache.version"/>
             <argument type="service" id="Shivas\VersioningBundle\Writer\WriterInterface"/>
             <argument type="service" id="Shivas\VersioningBundle\Formatter\FormatterInterface"/>
         </service>
 
-        <service id="Shivas\VersioningBundle\Formatter\GitDescribeFormatter"/>
-        <service id="Shivas\VersioningBundle\Writer\VersionWriter">
+        <service id="Shivas\VersioningBundle\Formatter\FormatterInterface" alias="shivas_versioning.formatter.git" public="false"/>
+        <service id="Shivas\VersioningBundle\Writer\WriterInterface" alias="shivas_versioning.writer.version" public="false"/>
+
+        <service id="shivas_versioning.formatter.git" class="Shivas\VersioningBundle\Formatter\GitDescribeFormatter" public="false"/>
+        <service id="shivas_versioning.writer.version" class="Shivas\VersioningBundle\Writer\VersionWriter" public="false">
             <argument>%kernel.project_dir%</argument>
         </service>
 
-        <service id="Shivas\VersioningBundle\Provider\VersionProvider">
+        <service id="shivas_versioning.provider.version" class="Shivas\VersioningBundle\Provider\VersionProvider" public="false">
             <argument>%kernel.project_dir%</argument>
             <tag name="shivas_versioning.provider" alias="version" priority="0"/>
         </service>
 
-        <service id="Shivas\VersioningBundle\Provider\GitRepositoryProvider">
+        <service id="shivas_versioning.provider.git" class="Shivas\VersioningBundle\Provider\GitRepositoryProvider" public="false">
             <argument>%kernel.project_dir%</argument>
             <tag name="shivas_versioning.provider" alias="git" priority="-25"/>
         </service>
 
-        <service id="Shivas\VersioningBundle\Provider\RevisionProvider">
+        <service id="shivas_versioning.provider.revision" class="Shivas\VersioningBundle\Provider\RevisionProvider" public="false">
             <argument>%kernel.project_dir%</argument>
             <tag name="shivas_versioning.provider" alias="revision" priority="-50"/>
         </service>
 
-        <service id="Shivas\VersioningBundle\Provider\InitialVersionProvider">
+        <service id="shivas_versioning.provider.init" class="Shivas\VersioningBundle\Provider\InitialVersionProvider" public="false">
             <tag name="shivas_versioning.provider" alias="init" priority="-75"/>
         </service>
 
-        <service id="Shivas\VersioningBundle\Command\StatusCommand">
+        <service id="shivas_versioning.cache.version" parent="cache.system" public="false">
+            <tag name="cache.pool" />
+        </service>
+
+        <service id="shivas_versioning.command.status" class="Shivas\VersioningBundle\Command\StatusCommand" public="false">
```

Comment:
you could use file-level defaults to mark all services as private, as the bundle requires 3.4+ anyway